devimint/src/util.rs

```diff
@@ -193,39 +193,64 @@ impl Command {
             .join(" ")
     }
 
-    /// Run the command and get its output as json.
+    /// Run the command and get its output string.
     pub async fn out_string(&mut self) -> Result<String> {
         let output = self
-            .run_inner()
+            .run_inner(false)
+            .await
+            .with_context(|| format!("command: {}", self.command_debug()))?;
+        let output = String::from_utf8(output.stdout)?;
+        Ok(output.trim().to_owned())
+    }
+
+    /// Run the command and get its output string, capturing stderr.
+    pub async fn out_string_capture_stderr(&mut self) -> Result<String> {
```

Comment:
@dpc re: https://github.com/fedimint/fedimint/pull/4391#issuecomment-1992895340

Let me know if you're okay with this approach. I agree we shouldn't need stderr too often and this seemed minimally invasive. I do like having the extra confidence the double spend is failing for the expected reason instead of an unrelated error.

Reply:
Yeah, it's fine for now.

But I feel like we're kind of inconsistent in what we are doing.

If we have a tool (`fedimint-cli`) that outputs json ... shouldn't it output json on `stderr` with an error as well?

It could print both human readable thing on `stderr` and json output on stderr that includes `error` field. 

@maan2003 What do you think?

Reply:
vote for json output with error on stdout, and non zero exit code.
and human readable output on stderr

Reply:
That's a good idea :+1: 

For now I've removed from this PR and will address in a followup https://github.com/fedimint/fedimint/issues/4558